src/sha256.ts

```diff
@@ -0,0 +1,5 @@
+import { sha256 as rawSha256 } from "@noble/hashes/sha256"
+import type { Input } from "@noble/hashes/utils"
+import { Effect } from "effect"
+
+export const sha256 = (input: Input) => Effect.succeed(rawSha256(input))
```

Comment:
This needs to use `Effect.sync`. Otherwise, you are not actually wrapping the call to `rawSha256` but instead the returned value.

Reply:
Furthermore, if this can fail, it should be using `Effect.try...`

Reply:
Even better, use `Either.try...` since this is synchronous.

`Either` is already `Effectable` so it can be used with the runtime seamlessly.